src/utilities/schemaHandlers.js

```diff
@@ -38,14 +38,26 @@ const preProcessHelper = (obj) => {
 
       // add default titles to dropdown of allowed types/ subschemas
       if (prop.anyOf) {
-        Object.values(prop.anyOf).forEach(option => {
-          // if the allowed type/ subschema is not a ref nor has a title,
-          // it defaults to <parentProp.title> option 1, <prop.title> option 2, ...
-          // we can convert to display the type name instead
-          if (!option.$ref && option.type && !option.title) {
-            option.title = option.type
-          }
-        })
+        const options = Object.values(prop.anyOf)
```

Comment:
nit: we can optimize this a little to avoid doing multiple joins, example:

```js
const options = Object.values(prop.anyOf).map(option => option.type).sort().join()
if (options === 'number,string') {
  prop.anyOf = [{ title: 'decimal', type: 'string', pattern: '^-?\\d+(\\.\\d{1,})?$' }]
} else if (options === 'null,number,string') {
  prop.anyOf = [{ type: 'null', title: 'null' }, { title: 'decimal', type: 'string', pattern: '^-?\\d+(\\.\\d{1,})?$' }]
} else {
  ...
```